x/reactions/keeper/alias_functions.go

```diff
@@ -172,7 +172,7 @@ func (k Keeper) IteratePostReactions(ctx sdk.Context, subspaceID uint64, postID
 func (k Keeper) HasReacted(ctx sdk.Context, subspaceID uint64, postID uint64, user string, value types.ReactionValue) bool {
 	found := false
 	k.IteratePostReactions(ctx, subspaceID, postID, func(reaction types.Reaction) (stop bool) {
-		if reaction.Author == user && reaction.Value.GetCachedValue().(types.ReactionValue) == value {
+		if reaction.Author == user && reaction.Value.GetCachedValue().(types.ReactionValue).Equal(value) {
```

Comment:
Using "==" always returns false here since `types.ReactionValue` is a interface

Reply:
Does this fix mean we might have double reactions stored on chain? Should be create a migration to make sure those are deleted if that's the case?

Reply:
It is, added the migration scripts and its test.